app/src/main/java/com/amaze/filemanager/ui/dialogs/GeneralDialogCreation.java

```diff
@@ -922,32 +926,61 @@ public static void showCompressDialog(final MainActivity m, final ArrayList<Hybr
     }
 
     public static void showSortDialog(final MainFragment m, AppTheme appTheme, final SharedPreferences sharedPref) {
+        final String path = m.getCurrentPath();
         int accentColor = m.getMainActivity().getAccent();
         String[] sort = m.getResources().getStringArray(R.array.sortby);
-        int current = Integer.parseInt(sharedPref.getString("sortby", "0"));
+        int current = SortHandler.getSortType(m.getContext(), path);
         MaterialDialog.Builder a = new MaterialDialog.Builder(m.getActivity());
         a.theme(appTheme.getMaterialDialogTheme());
         a.items(sort).itemsCallbackSingleChoice(current > 3 ? current - 4 : current,
                 (dialog, view, which, text) -> true);
+        final Set<String> onlyThisFloders = sharedPref.getStringSet(PREFERENCE_SORTBY_ONLY_THIS, new HashSet<>());
```

Comment:
> <p>Note that you <em>must not</em> modify the set instance returned
> by this call.  The consistency of the stored data is not guaranteed
> if you do, nor is your ability to modify the instance at all.

-- [source](https://developer.android.com/reference/android/content/SharedPreferences.html#getStringSet(java.lang.String,%20java.util.Set%3Cjava.lang.String%3E))

I like your use of Set for efficiently finding if Strings exist.